### PR TITLE
[1LP][RFR] Appliance provision notebook and miq scripts

### PR DESCRIPTION
**Pull request by lcouzens**
Added appliance provision and configuration for advanced environments to miq appliance scripts and notebooks for further customization.

### CODE REVIEW

File: cfme/scripting/appliance.py [hunk 1]
```diff
@@ -7,11 +7,15 @@
...
-
+import sys
 import click
 from cached_property import cached_property
 from functools import partial
-
+from collections import namedtuple
+from utils.appliance import get_or_create_current_appliance
+from utils.conf import credentials, cfme_data
+from cfme.test_framework.sprout.client import SproutClient
```
Comment (otsuman) [inline cfme/scripting/appliance.py]: REQUIRED: put imports in alphabetical order.

File: cfme/scripting/setup_env.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        count=count, preconfigured=False, lease_time=lease_time, provider=provider)
+    return apps
+
+
+@click.command('setup-env', help='Provisions and configures appliances')
+@click.option('--cfme-version', required=True)
+@click.option('--lease-time', default=180)
+@click.option('--ha', 'mode', flag_value='ha')
+@click.option('--distributed', 'mode', flag_value='distributed')
+@click.option('--replication', 'mode', flag_value='replication')
```
Comment (jkrocil) [inline cfme/scripting/setup_env.py]: `ha`, `distributed` and `replication` shouldn't be options, they should be a group commands. See `cfme/scripting/appliance.py`

File: cfme/scripting/setup_env.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        apps[1].wait_for_evm_service()
+        apps[1].wait_for_web_ui()
+        print("Remote region appliance provisioned and configured {}".format(ip1))
+        print("Setup - Replication on remote appliance")
+        apps[1].set_pglogical_replication(replication_type=':remote')
+        print("Setup - Replication on global appliance")
+        apps[0].set_pglogical_replication(replication_type=':global')
+        apps[0].add_pglogical_replication_subscription(apps[1].address)
+        print("Done!")
+    else:
```
[the remark this answers is not in the document]
Comment (lcouzens) [reply cfme/scripting/setup_env.py]: No we where going with Pete's checkout appliance I believe, Did we want Pete to add don't deamonise option? @psav

[the remark this answers is not in the document]
Comment (mshriver) [reply cfme/scripting/setup_env.py]: Do we have resolution on this request?

File: notebooks/Basic appliance setup.ipynb [hunk 4]
```diff
@@ -0,0 +1,186 @@
+{
```
Comment (mshriver) [inline notebooks/Basic appliance setup.ipynb]: Can we rename this without spaces?

**Pull request merged**